backend/src/main/java/ropold/backend/controller/RoomController.java

```diff
@@ -31,13 +34,23 @@ public List<RoomModel> getUserFavorites(@PathVariable String userId) {
 
     @PostMapping("/favorites/{userId}/{roomId}")
     @ResponseStatus(HttpStatus.CREATED)
-    public void addRoomToFavorites(@PathVariable String userId, @PathVariable String roomId) {
+    public void addRoomToFavorites(@PathVariable String userId, @PathVariable String roomId , @AuthenticationPrincipal OAuth2User authentication) {
+        String authenticatedUserId = authentication.getName();
+
+        if (!authenticatedUserId.equals(userId)) {
+            throw new AccessDeniedException("Access denied: User is not authorized to add this room to favorites.");
+        }
         appUserService.addRoomToFavorites(userId, roomId);
```

Comment:
Ich würde die userId aus der URL rauslassen. Du kannst diese Information einfach über den eingeloggten user abfragen.

```suggestion
@PostMapping("/favorites/{roomId}")
    @ResponseStatus(HttpStatus.CREATED)
    public void addRoomToFavorites(@PathVariable String roomId , @AuthenticationPrincipal OAuth2User authentication) {
        String authenticatedUserId = authentication.getName();
        appUserService.addRoomToFavorites(userId, roomId);
```